pkg/loop/internal/capabilities_registry.go

```diff
@@ -0,0 +1,667 @@
+package internal
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	"google.golang.org/grpc"
+
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/loop/internal/pb"
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+	valuespb "github.com/smartcontractkit/chainlink-common/pkg/values/pb"
+)
+
+// In the long-term, registration of capabilities should happen entirely
+// inside the core node, via well-defined registration paths. The canonical
+// example of this would be the standard capability registration path defined in
+// this design document:
+// https://docs.google.com/document/d/1i2m1BEytNUR-yBU5mLFVQLGv9FFgYWkVUllFGB062UQ/edit#heading=h.4e7ng0tekbkc
+
+// In the medium term, we need to define an OCR3 capability, and the best way
+// to do this is to pass in a capability registry when creating reporting plugin
+// loop.
```

Comment:
So are we creating a gRPC interface for Capability Registry only as a stop-gap?
Is the Registry going to exist inside the core node or will it be a LOOPP as well?
Why do we need to pass the Registry to a Capability so it can "register itself"? Can't we loading mechanism add Registry entries when spawning a new LOOPP?

Reply:
I wouldn't describe it as a stop gap: the world we are describing here is the "ultimate" world where all capabilities are standard capabilities, and we don't have any non-standard ones like those based on relayers or OCR.

I think the registry is a first-class node concept so it makes sense to have it in the core node. We could consider loopifying it in the future, but since it's so tied to the core node I'm not clear on the benefits of that.